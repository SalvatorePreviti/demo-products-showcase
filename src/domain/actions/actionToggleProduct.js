import { TOGGLE_PRODUCT } from '../actionTypes'

export default function actionToggleProduct(product) {
  return { type: TOGGLE_PRODUCT, product }
}
