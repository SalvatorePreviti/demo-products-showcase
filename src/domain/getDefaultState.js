function getDefaultState() {
  return {
    categories: [],
    activeCategory: null,
    productFilter: '',
    filteredProducts: [],
    activeProductsById: {},
    loading: false,
    loadFailed: false
  }
}

export default getDefaultState
