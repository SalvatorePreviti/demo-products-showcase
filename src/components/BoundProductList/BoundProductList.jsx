import React from 'react'
import { connect } from 'react-redux'
import ProductList from '../../atoms/ProductList/ProductList'
import actionToggleProduct from '../../domain/actions/actionToggleProduct'

function mapStateToProps(state) {
  const { filteredProducts, activeProductsById } = state
  return {
    items: filteredProducts,
    activeProductsById: activeProductsById
  }
}

function mapDispatchToProps(dispatch) {
  return {
    onSelect(product) {
      dispatch(actionToggleProduct(product))
    }
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(React.memo(ProductList))
