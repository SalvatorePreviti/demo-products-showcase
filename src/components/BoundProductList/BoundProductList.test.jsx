import React from 'react'
import getDefaultState from '../../domain/getDefaultState'
import mockStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import BoundProductList from './BoundProductList'
import actionToggleProduct from '../../domain/actions/actionToggleProduct'

describe('BoundProductList', () => {
  const items = [{ id: '1', title: 'title1' }, { id: '2', title: 'title2' }]

  let root
  let store

  beforeAll(() => {
    store = mockStore()({
      ...getDefaultState(),
      filteredProducts: items
    })
    root = shallow(<BoundProductList store={store} />)
  })

  it('snapshots', () => {
    expect(root).toMatchSnapshot()
  })

  it('passes props', () => {
    expect(root.props()).toMatchObject({ items })
  })

  it('passes callbacks', () => {
    root.simulate('select', items[1])
    expect(store.getActions()).toEqual([actionToggleProduct(items[1])])
  })
})
