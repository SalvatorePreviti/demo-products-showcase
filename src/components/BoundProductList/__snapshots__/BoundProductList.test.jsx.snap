// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoundProductList snapshots 1`] = `
<ProductList
  activeProductsById={Object {}}
  items={
    Array [
      Object {
        "id": "1",
        "title": "title1",
      },
      Object {
        "id": "2",
        "title": "title2",
      },
    ]
  }
  onSelect={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
